src/Illuminate/Foundation/helpers.php

```diff
@@ -681,9 +681,10 @@ function rescue(callable $callback, $rescue = null, $report = true)
     /**
      * Resolve a service from the container.
      *
-     * @param  string  $name
-     * @param  array  $parameters
-     * @return mixed
+     * @param    class-string<T>  $name
```

Comment:
> This will not work if using string names

Maybe using this would solve at least "half" of the problems, i.e. those cases indeed using a class?
```php
      * @param    class-string<T>|string  $name
```